File: C-Sharp/frmPayoutGroupList.cs
using CrystalDecisions.CrystalReports.Engine;
using CrystalDecisions.Windows.Forms;
using DpSdkEngLib;
using DPSDKOPSLib;
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using System.Linq;
using System.Xml.Linq;
 // ERROR: Not supported in C#: OptionDeclaration
namespace _4PosBackOffice.NET
{
	internal partial class frmPayoutGroupList : System.Windows.Forms.Form
	{
		string gFilter;
		ADODB.Recordset gRS;
		string gFilterSQL;
		int gID;
		short gSection;
		bool gAll;
		string grpDelete;

		public int getItem()
		{
			cmdNew.Text = "&All";
			cmdNew.Visible = false;
			doSearch();
			this.ShowDialog();
			return gID;

		}

		private void getNamespace()
		{

		}

		private void cmdExit_Click(System.Object eventSender, System.EventArgs eventArgs)
		{
			this.Close();
		}

		private void cmdNamespace_Click()
		{
			My.MyProject.Forms.frmFilter.loadFilter(ref gFilter);
			getNamespace();
		}

		private void cmdNew_Click(System.Object eventSender, System.EventArgs eventArgs)
		{
			if (gSection) {
				//Exit form and set indicator to suggest all
				gID = -1;
			} else {
				My.MyProject.Forms.frmPayoutGroup.loadItem(ref 0);
			}
			doSearch();
		}


		private void DataList1_DblClick(System.Object eventSender, KeyPressEventArgs eventArgs)
		{
			if (cmdNew.Visible) {
				if (!string.IsNullOrEmpty(DataList1.BoundText)) {
					My.MyProject.Forms.frmPayoutGroup.loadItem(ref Convert.ToInt32(DataList1.BoundText));
				}
				doSearch();
			} else {
				//If DataList1.BoundText = "" Then
				//    gID = 0
				//Else
				//    gID = DataList1.BoundText
				//End If
				switch (gSection) {
					case 0:
						gID = Convert.ToInt32(DataList1.BoundText);
						this.Close();
						break;
				}
			}
		}
		private void DataList1_KeyPress(System.Object eventSender, KeyPressEventArgs eventArgs)
		{
			switch (eventArgs.KeyChar) {
				case Strings.ChrW(13):
					DataList1_DblClick(DataList1, new System.EventArgs());
					eventArgs.KeyChar = Strings.ChrW(0);
					break;
				case Strings.ChrW(27):
					this.Close();
					eventArgs.KeyChar = Strings.ChrW(0);
					break;
			}

		}
		private void DataList1_MouseDown(System.Object eventSender, MouseEventArgs eventArgs)
		{
			ADODB.Recordset rs = default(ADODB.Recordset);

			 // ERROR: Not supported in C#: OnErrorStatement

			if (!string.IsNullOrEmpty(DataList1.BoundText)) {
				rs = modRecordSet.getRS(ref "SELECT PayoutGroup_Name FROM PayoutGroup WHERE PayoutGroupID =" + Conversion.Val(DataList1.BoundText));
				if (Strings.UCase(Strings.Mid(rs.Fields("PayoutGroup_Name").Value, 1, 8)) == "HANDHELD") {
					grpDelete = Strings.Trim(rs.Fields("PayoutGroup_Name").Value);
					if (eventArgs.Button == System.Windows.Forms.MouseButtons.Right) {
						//UPGRADE_ISSUE: Form method frmPayoutGroupList.PopupMenu was not upgraded. Click for more: 'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="CC4C7EC0-C903-48FC-ACCC-81861D12DA4A"'
						//Call PopupMenu(mnuHand)

						return;
					}
				}
			}

		}

		private void frmPayoutGroupList_KeyDown(System.Object eventSender, System.Windows.Forms.KeyEventArgs eventArgs)
		{
			short KeyCode = eventArgs.KeyCode;
			short Shift = eventArgs.KeyData / 0x10000;
			if (KeyCode == 36) {
				gAll = !gAll;
				doSearch();
				KeyCode = false;
			}
		}

		private void frmPayoutGroupList_KeyPress(System.Object eventSender, System.Windows.Forms.KeyPressEventArgs eventArgs)
		{
			short KeyAscii = Strings.Asc(eventArgs.KeyChar);
			switch (KeyAscii) {
				case System.Windows.Forms.Keys.Escape:
					KeyAscii = 0;
					cmdExit_Click(cmdExit, new System.EventArgs());
					break;
			}

			eventArgs.KeyChar = Strings.Chr(KeyAscii);
			if (KeyAscii == 0) {
				eventArgs.Handled = true;
			}
		}
		public void loadItem(ref short section)
		{
			gSection = section;
			if (gSection) {
				cmdNew.Text = "&All";
				cmdNew.Visible = false;
			}
			doSearch();
			this.ShowDialog();
		}
		public void loadItem1(ref short section)
		{
			gSection = section;
			if (gSection) {
				cmdNew.Text = "&All";
				cmdNew.Visible = false;
			}
			doSearch();

		}

		private void frmPayoutGroupList_FormClosed(System.Object eventSender, System.Windows.Forms.FormClosedEventArgs eventArgs)
		{
			gRS.Close();
		}

		public void mnuDel_Click(System.Object eventSender, System.EventArgs eventArgs)
		{
			short intres = 0;

			intres = Interaction.MsgBox("Do you want to delete this 'HANDHELD SCANNER' group", MsgBoxStyle.ApplicationModal + MsgBoxStyle.Question + MsgBoxStyle.YesNo, _4PosBackOffice.NET.My.MyProject.Application.Info.Title);
			if (intres == MsgBoxResult.Yes) {
				modRecordSet.cnnDB.Execute("DROP TABLE " + grpDelete);
				modRecordSet.cnnDB.Execute("DELETE * FROM PayoutGroup WHERE PayoutGroup_Name ='" + grpDelete + "'");
				loadItem1(ref 2);
			}

		}

		private void txtSearch_Enter(System.Object eventSender, System.EventArgs eventArgs)
		{
			txtSearch.SelectionStart = 0;
			txtSearch.SelectionLength = 999;
		}

		private void txtSearch_KeyDown(System.Object eventSender, System.Windows.Forms.KeyEventArgs eventArgs)
		{
			short KeyCode = eventArgs.KeyCode;
			short Shift = eventArgs.KeyData / 0x10000;
			switch (KeyCode) {
				case 40:
					this.DataList1.Focus();
					break;
			}
		}

		private void txtSearch_KeyPress(System.Object eventSender, System.Windows.Forms.KeyPressEventArgs eventArgs)
		{
			short KeyAscii = Strings.Asc(eventArgs.KeyChar);
			switch (KeyAscii) {
				case 13:
					doSearch();
					KeyAscii = 0;
					break;
			}
			eventArgs.KeyChar = Strings.Chr(KeyAscii);
			if (KeyAscii == 0) {
				eventArgs.Handled = true;
			}
		}

		private void doSearch()
		{
			string sql = null;
			string lString = null;
			lString = Strings.Trim(txtSearch.Text);
			lString = Strings.Replace(lString, "  ", " ");
			lString = Strings.Replace(lString, "  ", " ");
			lString = Strings.Replace(lString, "  ", " ");
			lString = Strings.Replace(lString, "  ", " ");
			lString = Strings.Replace(lString, "  ", " ");
			lString = Strings.Replace(lString, "  ", " ");
			if (string.IsNullOrEmpty(lString)) {
			} else {
				lString = "WHERE (PayoutGroup_Name LIKE '%" + Strings.Replace(lString, " ", "%' AND PayoutGroup_Name LIKE '%") + "%')";

			}
			if (gAll) {
			} else {
				if (string.IsNullOrEmpty(lString)) {
					lString = " WHERE PayoutGroup_Disabled = 0 ";
				} else {
					lString = lString + " AND PayoutGroup_Disabled = 0 ";
				}
			}

			gRS = modRecordSet.getRS(ref "SELECT DISTINCT PayoutGroupID, PayoutGroup_Name FROM PayoutGroup " + lString + " ORDER BY PayoutGroup_Name");

			//Display the list of Titles in the DataCombo
			DataList1.DataSource = gRS;
			DataList1.listField = "PayoutGroup_Name";

			//Bind the DataCombo to the ADO Recordset
			//UPGRADE_ISSUE: VBControlExtender property DataList1.DataSource is not supported at runtime. Click for more: 'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="74E732F3-CAD8-417B-8BC9-C205714BB4A7"'
			DataList1.DataSource = gRS;
			DataList1.boundColumn = "PayoutGroupID";

		}
	}
}

File: VB/4PosBackOffice.NET/frmPayoutGroupList.vb
Option Strict Off
Option Explicit On
Friend Class frmPayoutGroupList
	Inherits System.Windows.Forms.Form
	Dim gFilter As String
	Dim gRS As ADODB.Recordset
	Dim gFilterSQL As String
	Dim gID As Integer
	Dim gSection As Short
	Dim gAll As Boolean
	Dim grpDelete As String
	
	Public Function getItem() As Integer
		cmdNew.Text = "&All"
		cmdNew.Visible = False
		doSearch()
		Me.ShowDialog()
		getItem = gID
		
	End Function
	
	Private Sub getNamespace()
		
	End Sub
	
	Private Sub cmdExit_Click(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles cmdExit.Click
		Me.Close()
	End Sub
	
	Private Sub cmdNamespace_Click()
		frmFilter.loadFilter(gFilter)
		getNamespace()
	End Sub
	
	Private Sub cmdNew_Click(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles cmdNew.Click
		If gSection Then
			'Exit form and set indicator to suggest all
			gID = -1
		Else
			frmPayoutGroup.loadItem(0)
		End If
		doSearch()
	End Sub
	
    Private Sub DataList1_DblClick(ByVal eventSender As System.Object, ByVal eventArgs As KeyPressEventArgs) Handles DataList1.DoubleClick

        If cmdNew.Visible Then
            If DataList1.BoundText <> "" Then
                frmPayoutGroup.loadItem(CInt(DataList1.BoundText))
            End If
            doSearch()
        Else
            'If DataList1.BoundText = "" Then
            '    gID = 0
            'Else
            '    gID = DataList1.BoundText
            'End If
            Select Case gSection
                Case 0
                    gID = CInt(DataList1.BoundText)
                    Me.Close()
            End Select
        End If
    End Sub
    Private Sub DataList1_KeyPress(ByVal eventSender As System.Object, ByVal eventArgs As KeyPressEventArgs) Handles DataList1.KeyPress
        Select Case eventArgs.KeyChar
            Case ChrW(13)
                DataList1_DblClick(DataList1, New System.EventArgs())
                eventArgs.KeyChar = ChrW(0)
            Case ChrW(27)
                Me.Close()
                eventArgs.KeyChar = ChrW(0)
        End Select

    End Sub
    Private Sub DataList1_MouseDown(ByVal eventSender As System.Object, ByVal eventArgs As MouseEventArgs) Handles DataList1.MouseDown
        Dim rs As ADODB.Recordset

        On Error Resume Next
        If DataList1.BoundText <> "" Then
            rs = getRS("SELECT PayoutGroup_Name FROM PayoutGroup WHERE PayoutGroupID =" & Val(DataList1.BoundText))
            If UCase(Mid(rs.Fields("PayoutGroup_Name").Value, 1, 8)) = "HANDHELD" Then
                grpDelete = Trim(rs.Fields("PayoutGroup_Name").Value)
                If eventArgs.Button = Windows.Forms.MouseButtons.Right Then
                    'UPGRADE_ISSUE: Form method frmPayoutGroupList.PopupMenu was not upgraded. Click for more: 'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="CC4C7EC0-C903-48FC-ACCC-81861D12DA4A"'
                    'Call PopupMenu(mnuHand)

                    Exit Sub
                End If
            End If
        End If

    End Sub
	
	Private Sub frmPayoutGroupList_KeyDown(ByVal eventSender As System.Object, ByVal eventArgs As System.Windows.Forms.KeyEventArgs) Handles MyBase.KeyDown
		Dim KeyCode As Short = eventArgs.KeyCode
		Dim Shift As Short = eventArgs.KeyData \ &H10000
		If KeyCode = 36 Then
			gAll = Not gAll
			doSearch()
			KeyCode = False
		End If
	End Sub
	
	Private Sub frmPayoutGroupList_KeyPress(ByVal eventSender As System.Object, ByVal eventArgs As System.Windows.Forms.KeyPressEventArgs) Handles MyBase.KeyPress
		Dim KeyAscii As Short = Asc(eventArgs.KeyChar)
		Select Case KeyAscii
			Case System.Windows.Forms.Keys.Escape
				KeyAscii = 0
				cmdExit_Click(cmdExit, New System.EventArgs())
		End Select
		
		eventArgs.KeyChar = Chr(KeyAscii)
		If KeyAscii = 0 Then
			eventArgs.Handled = True
		End If
	End Sub
	Public Sub loadItem(ByRef section As Short)
		gSection = section
		If gSection Then
			cmdNew.Text = "&All"
			cmdNew.Visible = False
		End If
		doSearch()
		Me.ShowDialog()
	End Sub
	Public Sub loadItem1(ByRef section As Short)
		gSection = section
		If gSection Then
			cmdNew.Text = "&All"
			cmdNew.Visible = False
		End If
		doSearch()
		
	End Sub
	
	Private Sub frmPayoutGroupList_FormClosed(ByVal eventSender As System.Object, ByVal eventArgs As System.Windows.Forms.FormClosedEventArgs) Handles Me.FormClosed
		gRS.Close()
	End Sub
	
	Public Sub mnuDel_Click(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles mnuDel.Click
		Dim intres As Short
		
		intres = MsgBox("Do you want to delete this 'HANDHELD SCANNER' group", MsgBoxStyle.ApplicationModal + MsgBoxStyle.Question + MsgBoxStyle.YesNo, My.Application.Info.Title)
		If intres = MsgBoxResult.Yes Then
			cnnDB.Execute("DROP TABLE " & grpDelete)
			cnnDB.Execute("DELETE * FROM PayoutGroup WHERE PayoutGroup_Name ='" & grpDelete & "'")
			loadItem1(2)
		End If
		
	End Sub
	
	Private Sub txtSearch_Enter(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles txtSearch.Enter
		txtSearch.SelectionStart = 0
		txtSearch.SelectionLength = 999
	End Sub
	
	Private Sub txtSearch_KeyDown(ByVal eventSender As System.Object, ByVal eventArgs As System.Windows.Forms.KeyEventArgs) Handles txtSearch.KeyDown
		Dim KeyCode As Short = eventArgs.KeyCode
		Dim Shift As Short = eventArgs.KeyData \ &H10000
		Select Case KeyCode
			Case 40
				Me.DataList1.Focus()
		End Select
	End Sub
	
	Private Sub txtSearch_KeyPress(ByVal eventSender As System.Object, ByVal eventArgs As System.Windows.Forms.KeyPressEventArgs) Handles txtSearch.KeyPress
		Dim KeyAscii As Short = Asc(eventArgs.KeyChar)
		Select Case KeyAscii
			Case 13
				doSearch()
				KeyAscii = 0
		End Select
		eventArgs.KeyChar = Chr(KeyAscii)
		If KeyAscii = 0 Then
			eventArgs.Handled = True
		End If
	End Sub
	
	Private Sub doSearch()
		Dim sql As String
		Dim lString As String
		lString = Trim(txtSearch.Text)
		lString = Replace(lString, "  ", " ")
		lString = Replace(lString, "  ", " ")
		lString = Replace(lString, "  ", " ")
		lString = Replace(lString, "  ", " ")
		lString = Replace(lString, "  ", " ")
		lString = Replace(lString, "  ", " ")
		If lString = "" Then
		Else
			lString = "WHERE (PayoutGroup_Name LIKE '%" & Replace(lString, " ", "%' AND PayoutGroup_Name LIKE '%") & "%')"
			
		End If
		If gAll Then
		Else
			If lString = "" Then
				lString = " WHERE PayoutGroup_Disabled = 0 "
			Else
				lString = lString & " AND PayoutGroup_Disabled = 0 "
			End If
		End If
		
		gRS = getRS("SELECT DISTINCT PayoutGroupID, PayoutGroup_Name FROM PayoutGroup " & lString & " ORDER BY PayoutGroup_Name")
		
		'Display the list of Titles in the DataCombo
        DataList1.DataSource = gRS
		DataList1.listField = "PayoutGroup_Name"
		
		'Bind the DataCombo to the ADO Recordset
		'UPGRADE_ISSUE: VBControlExtender property DataList1.DataSource is not supported at runtime. Click for more: 'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="74E732F3-CAD8-417B-8BC9-C205714BB4A7"'
		DataList1.DataSource = gRS
		DataList1.boundColumn = "PayoutGroupID"
		
	End Sub
End Class